app/src/main/java/com/example/jetpackcompose/state/ProcessDeathActivity.kt

```diff
@@ -0,0 +1,184 @@
+                    // CreditCardVisualTransformation. All it does is that it transforms the input
+                    // to have a space added after every 4 characters. The text itself isn't
+                    // altered, just its visual appearance is. You can easily created you own visual
+                    // transformations by implementing the VisualTransformation interface like we did
+                    // for CreditCardVisualTransformation.
+                    visualTransformation = CreditCardVisualTransformation(),
+                    // Update value of textValue with the latest value of the text field
+                    onValueChange = { newTextValue ->
+                        textValue = newTextValue
+                    }
+                )
+                // Text is a predefined composable that does exactly what you'd expect it to -
+                // display text on the screen. It allows you to customize its appearance using
+                // the style property.
+                Text(
+                    text = "John Doe",
+                    modifier = Modifier.padding(16.dp),
+                    style = TextStyle(
+                        color = Color.White,
+                        fontSize = 25.sp,
+                        fontFamily = FontFamily.Cursive,
+                        fontWeight = FontWeight.Bold
+                    )
+                )
+            }
+        }
+    }
+}
+
+// Visual transformation is used to modify the visual output of the input field. In this example,
+// I'm using a custom visual transformation - the CreditCardVisualTransformation. All it does is
+// that it transforms the input to have a space added after every 4 characters. The text itself isn't
+// altered, just its visual appearance is. You can easily created you own visual transformations by
+// implementing the VisualTransformation interface like we did for CreditCardVisualTransformation.
+class CreditCardVisualTransformation: VisualTransformation {
+    override fun filter(text: AnnotatedString): TransformedText {
+        return TransformedText(
+            // Regex used to add a space after every 4 characters.
+            AnnotatedString(text.text.replace("....".toRegex(), "$0 ")),
+            creditCardOffsetMap
+        )
+    }
+}
+
+// Offset map is used to change the position of the cursor based on the transormation being
+// applied. If no offset is needed for the cursor position, return the same offset value. In the
+// case of CreditCardVisualTransformation, since we add a space character after every 4
+// characters, we need to move the cursor accordingly. For example, if we added 3 space
+// characters, we need to account for that and move the cursor offset by 3 characters.
+val creditCardOffsetMap =  object: OffsetMap {
+    override fun originalToTransformed(offset: Int) = offset + (offset/4)
+    override fun transformedToOriginal(offset: Int) = offset - (offset/4)
```

Comment:
Need to update this logic to work perfectly, fine for now since the purpose of this example is not this part.